custom_components/google_home/sensor.py

```diff
@@ -143,6 +145,12 @@ def _get_alarms_data(self) -> List[Dict[Any, Any]]:
         """Update alarms data extracting it from coordinator"""
         device = self.get_device()
         return self.as_dict(device.get_sorted_alarms()) if device else []
+    
+    def _get_next_alarm_status(self) -> str:
+        """Update next alarm status from coordinator"""
+        device = self.get_device()
+        next_alarm = device.get_next_alarm()
```

Comment:
Return `STATUS_NONE` if `device is None`.

Reply:
Could we differentiate between the device reporting *no status* and the device being unreachable?
E.g. If my network has a hiccup at night and the device is temporarily not available, I do not want my wake-up automation to go off. But otherwise triggering on a state change from "ringing" to "none" would be super convenient.

Maybe add an additional STATUS_UNAVAILABLE? Or just not update the status and cache it?

Reply:
This for status of individual alarm not state of entity. 

But right now we have an issue with using `unavailable` for state when there are no alarms set. This is because we need to use sensor device class `timestamp` which only support such state when no actual timestamp is set. This causes entities appearing unavailable on the list of entities defined by integration. I think we can override `is_available` method of sensor entity and set it manually.